crates/compilers/src/compilers/multi.rs

```diff
@@ -288,11 +303,27 @@ impl Compiler for MultiCompiler {
     ) -> Result<CompilerOutput<Self::CompilationError, Self::CompilerContract>> {
         match input {
             MultiCompilerInput::Solc(input) => {
-                if let Some(solc) = &self.solc {
-                    Compiler::compile(solc, input).map(|res| res.map_err(MultiCompilerError::Solc))
-                } else {
-                    Err(SolcError::msg("solc compiler is not available"))
+                if self.use_resolc {
+                    if let Some(resolc) = &self.resolc {
+                        let input = input.clone();
+                        let input = ResolcVersionedInput::build(
+                            input.input.sources,
+                            SolcSettings {
+                                settings: input.input.settings,
+                                cli_settings: input.cli_settings,
+                            },
+                            input.input.language,
+                            input.version,
+                        );
+                        return Compiler::compile(resolc, &input)
+                            .map(|res| res.map_err(MultiCompilerError::Solc));
+                    }
+                } else if let Some(solc) = &self.solc {
+                    return Compiler::compile(solc, input)
+                        .map(|res| res.map_err(MultiCompilerError::Solc));
                 }
+
+                Err(SolcError::msg("resolc compiler is not available"))
```

Comment:
I think we should change to `revive compiler is not available`

Reply:
the compiler is explicitly named `resolc`, @smiasojed what's your opinion on this?

Reply:
Ok, Im just concerned about inconsistent naming, on foundry we use `revive-compile` and `revive_path` then when we get errors we mentioning `resolc`